### ISSUE TITLE
Unable to get out of code and citation blocks in a conversation

### ISSUE THREAD

**Issue by il2020** [freescout-help-desk/freescout#3811]
**Steps to reproduce:**
1. Hit Reply, write something (one word) and hit Enter.
2. Insert a code or citation block.
3. Try to write something after it.

**Result:**
Unable to place the cursor after a block.

**Expected result:**
You can continue to write a reply.

PHP version: 8.2.7
FreeScout version: 1.8.120
Database: MySQL
Are you using CloudFlare: No

**Comment by DavidAnderson684** [comment]
I've always had this problem too - I've never got round to reporting it, but I instead write the part 'after' first (or rather, write a placeholder), and then put the code block in between.... then I can go to the 'after' part and carry on. It would be nice if it were fixed. Using Firefox.

**Comment by sparker81** [comment]
@freescout-helpdesk, could the answer be as simple as adding a <br> after Code or Quote? That way there's always a new line to move to once you've inserted a block.

**Comment by freescout-help** [comment]
Fixed in the `master` branch and will be published in the next release.

**Comment by freescout-help** [comment]
The fix for this issue has been reverted as it caused other issues:

- https://github.com/freescout-help-desk/freescout/issues/4528
- https://github.com/freescout-help-desk/freescout/issues/4520

If someone can find a proper solution please share it.

**Comment by ilyakonovalenko** [comment]
By the way, I am able to get out of code block at summernote.org with Option+Enter on Mac.

**Comment by anxodio** [comment]
For me summernote.org has a behaviour that I like... when I'm in a code or quote block:

- If i just press enter, it goes out
- If i want to insert a line inside the code/quote block, i need to press shift+enter (like other similar components)

**Comment by freescout-help** [comment]
Yes, the issue exists only FreeScout editor as it uses `div` instead of `p` to separate paragraphs.

**Comment by ilyakonovalenko** [comment]
Is there any specific reason to use `div` instead of `p`?

**Comment by freescout-help** [comment]
> Is there any specific reason to use `div` instead of `p`?

There were issues with displaying `<p>` tags in some email clients as they apply different default styles to `<p>` while `<div>` has no default styles applied.

**Comment by anxodio** [comment]
Now, in my freescout (v1.8.161), it works like:
- For a citation block, if i just press enter, i get another line in the citation block. If I press enter again, I get out of the citation block (easy way to exit!).
- Also for a citation block, if i press shift+enter, i get out.
- And now I have a strange behaviour in code block... if I create a code block, write something and press enter, it goes out of the code block, i can't easily create code blocks of more than one line...

The point is that the double enter trick for me it's ok for exit of a block (it can be in a helptool or something), and I don't know whats happening with the code blocks, anyone else is having this issue?

**Comment by il2020** [comment]
> anyone else is having this issue?

Yeah, I have the same issue.

**Comment by il2020** [comment]
Why did you close this issue? In version 1.8.170, the problem remains. I press both enter and shift+enter inside the code block, but I can't get out of it.

![Image](https://github.com/user-attachments/assets/92f2fcbb-9908-4368-bef8-7ff4adde4f20)

**Comment by freescout-help** [comment]
> The fix for this issue has been reverted as it caused other issues:
> 
> * [Unable to shift+enter from a ordered/unordered list #4528](https://github.com/freescout-help-desk/freescout/issues/4528)
> * [New line gets added after the last line of the text in some cases #4520](https://github.com/freescout-help-desk/freescout/issues/4520)
> 
> If someone can find a proper solution please share it.

**Comment by anxodio** [comment]
Hi! I Just made a little PR that I thinks that solves all the issues (in my tests all look good).

Can you check it out? https://github.com/freescout-help-desk/freescout/pull/4607

**Comment by freescout-help** [comment]
@anxodio Thanks. It works fine. It has been merged into `master` branch and will be published in the next release.
